### ISSUE TITLE
Implement off-chain downstream to on-chain upstream incoming HTLC claiming

### ISSUE THREAD

**Issue by ariard** [rust-bitcoin/rust-lightning#595]
See TODO in ChannelManager

**Comment by TheBlueMatt** [comment]
Specifically, this TODO:
```
						// TODO: There is probably a channel monitor somewhere that needs to
						// learn the preimage as the channel already hit the chain and that's
						// why it's missing.
```

**Comment by ariard** [comment]
@valentinewallace closed by #611 ?

**Comment by valentinewallace** [comment]
Yup, I think that's right @ariard
